File: apps/web/src/features/agent/tool-icons.ts
import {
  AppWindow,
  Bot,
  FileCode,
  FileText,
  Globe,
  type LucideIcon,
  Monitor,
  NotebookPen,
  Plug,
  Search,
  Terminal,
  Wrench,
} from "lucide-react";

/** Icon by tool family (names from packages/agent/src/tools/contracts.ts). */
export function toolIcon(toolName: string): LucideIcon {
  if (toolName.startsWith("mcp__")) return Plug;
  if (toolName.startsWith("browser_")) return AppWindow;
  if (toolName.startsWith("computer_")) return Monitor;
  if (toolName === "web_search") return Search;
  if (toolName.startsWith("web_")) return Globe;
  if (toolName === "bash") return Terminal;
  if (["read", "write", "edit", "grep", "find", "ls"].includes(toolName)) return FileCode;
  if (toolName === "read_note" || toolName === "search_notes") return NotebookPen;
  if (toolName === "create_artifact" || toolName === "post_update") return FileText;
  if (["spawn_subagent", "message_subagent", "cancel_subagent", "ask_user"].includes(toolName))
    return Bot;
  return Wrench;
}

File: apps/macos/Packages/DailyDoListAgent/Sources/DailyDoListAgent/Support/ToolIcons.swift
/// SF Symbols by tool family (names from `packages/agent/src/tools/contracts.ts`).
public enum ToolIcon {
  public static func systemName(for toolName: String) -> String {
    if toolName.hasPrefix("mcp__") { return "puzzlepiece.extension" }
    if toolName.hasPrefix("browser_") { return "globe" }
    if toolName.hasPrefix("computer_") { return "cursorarrow.rays" }
    if toolName.hasPrefix("web_") { return "magnifyingglass" }
    switch toolName {
    case "bash": return "terminal"
    case "read": return "doc.text"
    case "write", "edit": return "square.and.pencil"
    case "grep", "find": return "doc.text.magnifyingglass"
    case "ls": return "folder"
    case "read_note", "search_notes": return "note.text"
    case "read_drawing": return "scribble.variable"
    case "post_update", "post_comment": return "text.bubble"
    case "ask_user": return "questionmark.bubble"
    case "create_artifact": return "doc.richtext"
    case "finish_task", "set_task_status": return "checklist"
    case "list_tasks": return "list.bullet"
    case "spawn_subagent", "message_subagent", "cancel_subagent": return "person.2"
    case "mock_irreversible_action": return "exclamationmark.triangle"
    default: return "wrench.and.screwdriver"
    }
  }
}
